clinica/pipelines/dwi_connectome/dwi_connectome_pipeline.py

```diff
@@ -64,118 +65,124 @@ def build_input_node(self):
-                                    extensions='nii.gz',
-                                    subject=self.subjects[i][4:],
-                                    return_type='id')[0])
-
-            if dwi_file_space[i] == 'b0':
-                t1_brain_file.append(
-                        caps_layout.get(freesurfer_file='brain.mgz',
-                                        session=self.sessions[i][4:],
-                                        subject=self.subjects[i][4:],
-                                        return_type='file')[0])
-
-            aparc_aseg.append(
-                    caps_layout.get(freesurfer_file='aparc\+aseg.mgz',
-                                    session=self.sessions[i][4:],
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            aparc_aseg_a2009s.append(
-                    caps_layout.get(freesurfer_file='aparc.a2009s\+aseg.mgz',
-                                    session=self.sessions[i][4:],
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            atlas_files.append([aparc_aseg[i], aparc_aseg_a2009s[i]])
-
-            dwi_file.append(
-                    caps_layout.get(type='dwi', suffix='preproc',
-                                    space=dwi_file_space[i],
-                                    session=self.sessions[i][4:],
-                                    extensions='nii.gz',
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            dwi_brainmask_file.append(
-                    caps_layout.get(type='dwi', suffix='brainmask',
-                                    space=dwi_file_space[i],
-                                    session=self.sessions[i][4:],
-                                    extensions='nii.gz',
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            bvec.append(
-                    caps_layout.get(type='dwi', suffix='preproc',
-                                    space=dwi_file_space[i],
-                                    session=self.sessions[i][4:],
-                                    extensions='bvec',
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            bval.append(
-                    caps_layout.get(type='dwi', suffix='preproc',
-                                    space=dwi_file_space[i],
-                                    session=self.sessions[i][4:],
-                                    extensions='bval',
-                                    subject=self.subjects[i][4:],
-                                    return_type='file')[0])
-
-            grad_fsl.append((bvec[i], bval[i]))
+            # Inputs from t1-freesurfer pipeline
```

Comment:
Je pense que le code ci dessous n'est pas optimisé pour deux raisons :  
- tu itères sur tes n patient avec un `for` : tu appelles donc n * 10 fois caps_layout. Dans un dossier CAPS important, la requete peut vite prendre quelques secondes (quand je faisais ca dans pet-surface, avec un caps de 240 sujets, ca mettait une heure à attraper les fichiers). 
- tu effectues un check pour chaque fichier. Si l'utilisateur a un jeu de données incomplet, il va devoir relancer le pipeline et régler un à un les données manquantes. Je pense qu'il est plus judicieux de noter tout ce qu'il manque et de lancer une exception à la fin en explicitant tout. 

Pour le problème 1:
- Tu peux utiliser la librairie `glob` qui permet d'aller chercher des fichiers qui matchent un pattern (et il est possible facilement de faire un truc robuste à la casse)
- Faire comme dans `pet_surface_pipeline`: grabber avec caps_layout une seule fois par type de fichier (l'ordre n'est pas garanti d'etre le meme entre les différents fichier) et un peu de manip derrière pour les remettre dans le bon ordre 

Problème 2:
Transformer ta fonction de check pour qu'elle opère sur l'ensemble de tout les fichiers identifiés afin de présenter un "bilan" total à l'utilisateur

Reply:
Si ma nouvelle solution te convient, peux-tu mettre cette conversation en "Resolve" ?

Reply:
Il reste l'histoire de l'ordre des fichiers qui sont grabbés à régler

Reply:
J'ai tendance à penser que ce scénario n'arriverait jamais, sinon la fonction pour récupérer les fichiers est problématique.

Dans le doute, j'ai rajouté une vérification supplémentaire dans `check_input_caps_files` (commit fd616d1).

Reply:
Peux-tu mettre cette conversation à "Resolved" ? Comme on continue la conversation [plus bas](https://github.com/aramis-lab/clinica/pull/8#discussion_r318119873).